tools/diff-kernel-config

```
@@ -0,0 +1,240 @@
+
+    gitrev_var=gitrev_${state}
+    git checkout --quiet "${!gitrev_var}" || bail "Cannot check out '${!gitrev_var}'."
+
+    for arch in aarch64 x86_64; do
+
+        for kver in "${kernel_versions[@]}"; do
+
+            variants=()
+            case ${kver} in
+                5.10)
+                    variants+=( 'aws-k8s-1.23' )
+                    if [[ ${arch} = x86_64 ]]; then
+                        variants+=( 'metal-k8s-1.23' )
+                    fi
+                    ;;
+                5.15)
+                    variants+=( 'aws-dev' 'metal-dev' )
+                    ;;
+                *)
+                    bail "No known variants build kernel ${kver}."
+                    ;;
+            esac
+
+            for variant in "${variants[@]}"; do
+
+                debug_id="state=${state} arch=${arch} variant=${variant}"
+
+                #
+                # Run build
+                #
+
+                cargo make \
+                        -e BUILDSYS_ARCH="${arch}" \
+                        -e BUILDSYS_VARIANT="${variant}" \
+                        -e PACKAGE="kernel-${kver/./_}" \
+                        build-package \
+                    || bail "Build failed for ${debug_id}"
+
+                #
+                # Find kernel RPM
+                #
+
+                shopt -s nullglob
+                kernel_rpms=(
+                    ./build/rpms/bottlerocket-"${arch}"-*kernel-"${kver}"-"${kver}".*.rpm
+                )
+                shopt -u nullglob
+
+                case ${#kernel_rpms[@]} in
+                    0) bail "No kernel RPM found for ${debug_id}" ;;
+                    1) kernel_rpm=${kernel_rpms[0]} ;;
+                    *) bail "More than one kernel RPM found for ${debug_id}" ;;
+                esac
+
+
+                #
+                # Extract kernel config
+                #
+
+                mkdir -p "${output_dir}" || bail "Failed to create output directory '${output_dir}'"
```

I believe this is something we should do before we start any build. Would be a shame if we spend the time of a build to find out that we can not create the output directory for some reason.

---

Thankfully a single build is comparatively cheap. Agree, though, for obvious mistakes (e.g. no write permissions) we should fail early. Changed in the force push.